pynxtools/dataconverter/readers/multi/reader.py

```diff
@@ -0,0 +1,208 @@
+
+    return config_file_dict
+
+
+class MultiFormatReader(BaseReader):
+    """
+    A reader that takes a mapping json file and a data file/object to return a template.
+    """
+
+    # Whitelist for the NXDLs that the reader supports and can process
+    supported_nxdls: List[str] = []
+    extensions: Dict[str, Callable[[Any], dict]] = {
+        ".yaml": parse_yml,
+        ".yml": parse_yml,
+    }
+    kwargs: Optional[Dict[str, Any]] = None
+
+    def __init__(self):
+        self.extensions[".json"] = lambda fn: parse_json_config(
+            fn,
+            ParseJsonCallbacks(
+                attrs_callback=self.get_attr,
+                data_callback=self.get_data,
+                dims=self.get_data_dims,
+            ),
+        )
+
+    def setup_template(self) -> Dict[str, Any]:
+        """
+        Setups the initial data in the template.
+        This may be used to set fixed information, e.g., about the reader.
+        """
+        return {}
+
+    # pylint: disable=unused-argument
+    def handle_objects(self, objects: Tuple[Any]) -> Dict[str, Any]:
+        """
+        Handles the objects passed into the reader.
+        """
+        return {}
+
+    def get_attr(self, path: str) -> Any:
+        """
+        Returns an attributes from the given path.
+        """
+        return {}
+
+    def get_data(self, path: str) -> Any:
+        """
+        Returns data from the given path.
+        """
+        return {}
+
+    def get_data_dims(self, path: str) -> List[str]:
+        """
+        Returns the dimensions of the data from the given path.
+        """
+        return []
+
+    def read(
+        self,
+        template: dict = None,
+        file_paths: Tuple[str] = None,
+        objects: Tuple[Any] = None,
+        **kwargs,
+    ) -> dict:
+        """
+        Reads data from multiple files and passes them to the appropriate functions
+        in the extensions dict.
+        """
+        template = Template()
+        self.kwargs = kwargs
+
+        def sort_keys(filename: str) -> str:
+            """
+            Makes sure to read json and yaml files last
+            """
+            ending = os.path.splitext(filename)[1]
+            return ".zzzzz" if ending in (".json", ".yaml", ".yml") else ending
+
+        sorted_paths = sorted(file_paths, key=sort_keys)
+        for file_path in sorted_paths:
+            extension = os.path.splitext(file_path)[1].lower()
+            if extension not in self.extensions:
+                print(
+                    f"WARNING: "
+                    f"File {file_path} has an unsupported extension, ignoring file."
+                )
+                continue
+            if not os.path.exists(file_path):
+                print(f"WARNING: File {file_path} does not exist, ignoring entry.")
+                continue
+
+            template.update(self.extensions.get(extension, lambda _: {})(file_path))
```

Comment:
Alternative: I have a [function](https://github.com/FAIRmat-NFDI/pynxtools-xps/blob/c0788e5a8ba6e8c769d9ca956aded28d83dae512/pynxtools_xps/reader.py#L409) that only updates if there is nothing overwritten that already exists in the template.

Reply:
Let's discuss this. In my view it might be useful sometimes to allow overwriting, e.g., with the json dict. But I could also imagine giving the reader a flag to switch this behaviour and set one of the options as default.